src/d3d12/d3d12_heap.cpp

```diff
@@ -466,13 +466,14 @@ namespace wr::d3d12
 		return cb;
 	}
 
-	HeapResource * AllocStructuredBuffer(Heap<HeapOptimization::BIG_STATIC_BUFFERS>* heap, std::uint64_t size_in_bytes, std::uint64_t stride)
+	HeapResource * AllocStructuredBuffer(Heap<HeapOptimization::BIG_STATIC_BUFFERS>* heap, std::uint64_t size_in_bytes, std::uint64_t stride, bool used_as_uav)
 	{
 		auto cb = new HeapResource();
 		decltype(Device::m_native) n_device;
 		heap->m_native->GetDevice(IID_PPV_ARGS(&n_device));
 		cb->m_unaligned_size = size_in_bytes;
 		cb->m_stride = stride;
+		cb->m_used_as_uav = used_as_uav;
```

Comment:
Is this member variable necessary? I don't see it used anywhere beside this function.

Reply:
m_used_as_uav is needed to transition it properly after updating. The older implementation of UpdateStructuredBuffer assumed UAV, while that is almost never the case.